### ISSUE TITLE
Potential Denial of service on arbitrum-dai-bridge/contracts/l1/L1GovernanceRelay.sol contract

### ISSUE THREAD

**Issue by OceansAndThunders** [makerdao/arbitrum-dai-bridge#39]
The contract [L1GovernanceRelay](https://github.com/makerdao/arbitrum-dai-bridge/blob/60f665bfb6ef4f0258392e47d1ee9fd4b95f4ae7/contracts/l1/L1GovernanceRelay.sol) uses funds sending instead of withdrawal on the claim of stored ethers 

```
function reclaim(address receiver, uint256 amount) external auth {
    (bool sent, ) = receiver.call{value: amount}("");
    require(sent, "L1GovernanceRelay/failed-to-send-ether");
  }

```

This can lead to potential denial of service and locking the contract from working as intended permanently by calling calling the reclaim function as intended by without a fallback function to tells that the ethers has received, this will make the ruls `require(sent, "L1GovernanceRelay/failed-to-send-ether")` unsatisified, yet it won't process other actions neither because it still processing that one, and so on ...
the contract will freeze

an example exploit of the contract ;

```
contract exploit {
  function do(L1GovernanceRelay L1GovernanceRelay, address add, uint256 am) public payable {
       L1GovernanceRelay.reclaim(add, am);
       // this will makes the ethers received but with no fallback action it won't satisfy the "require(sent, ...etc" rule, and the process won't finish ever 
  }
}
```

To fix this ,replace the code by making the reclaim withdrawal not via sent, 
example 

```
function reclaim(address receiver, uint256 amount) external auth {
    require(msg.value > amount, "not enough");
    balances[receiver] += amount;
  }

```

This will lead to freeze all the ethers inside the contract, please take care on this
and sorry if this wasn't helpful as much as I thought so

Hope it helps

**Comment by krzkaczor** [comment]
Hey! Thanks for the report. My response is late but I assure you that we verified this right after you posted it. 

1. The method `reclaim` is authed and it can be only called via governance. [Related unit test](https://github.com/makerdao/arbitrum-dai-bridge/blob/master/test/l1/L1GovernanceRelay.ts#L109) 
2. 

> This can lead to potential denial of service and locking the contract from working as intended permanently by calling calling the reclaim function as intended by without a fallback function to tells that the ethers has received, this will make the ruls require(sent, "L1GovernanceRelay/failed-to-send-ether") unsatisified, yet it won't process other actions neither because it still processing that one, and so on ... the contract will freeze.

The `reclaim` function was written to be used with EOA in mind. But even when it's used with a contract address, the `relayer` contract won't "freeze" as you suggest but rather the whole call will revert. This will enable us to spot such an issue while testing the governance spell that would reclaim funds. 

To sum up, it can't be exploited. Thanks for looking into our contracts! Reopen if needed.
